src/impl_/pyclass.rs

```diff
@@ -811,8 +813,8 @@ slot_fragment_trait! {
         other: *mut ffi::PyObject,
     ) -> PyResult<*mut ffi::PyObject> {
         // By default `__ne__` will try `__eq__` and invert the result
-        let slf: &PyAny = py.from_borrowed_ptr(slf);
-        let other: &PyAny = py.from_borrowed_ptr(other);
+        let slf: Bound<'_,PyAny> = Bound::from_borrowed_ptr(py, slf);
+        let other: Bound<'_,PyAny> = Bound::from_borrowed_ptr(py, other);
```

Comment:
Here I would suggest using `Borrowed::from_ptr` to avoid the reference count operation. We're again on the surface of a function call from CPython so it's the interpreter's responsibility to make sure these objects live long enough.

Unlike above in the macro implementation, here there are no arbitrary user types involved so we can safely use `Borrowed::from_ptr` without needing to worry about constraining that wildly unsafe lifetime `'a`.